### PR TITLE
Fix Color constructor parameter types

### PR DESCRIPTION
**Pull request by latin-1**

### CODE REVIEW

File: types/src/color.d.ts [hunk 1]
```diff
@@ -22,9 +22,10 @@ import {
 export type Coords = [number, number, number];
 
 export interface ColorObject {
-	space: ColorSpace;
+	spaceId?: string | ColorSpace | undefined;
+	space?: string | ColorSpace | undefined;
 	coords: Coords;
-	alpha?: number;
+	alpha?: number | undefined;
```
Comment (LeaVerou) [inline types/src/color.d.ts:28]: Does it need `| undefined` if it's marked optional?

**Comment by LeaVerou** [comment]
LGTM, @MysteryBlokHed @jgerigmeyer do you know if optional keys still need `| undefined`?

**Comment by MysteryBlokHed** [comment]
It's a good idea to explicitly add `| undefined` to optional properties because of TypeScript's `exactOptionalPropertyTypes` config option, which treats optional properties differently than properties with a value of `undefined`: <https://devblogs.microsoft.com/typescript/announcing-typescript-4-4/#exact-optional-property-types>

**Comment by LeaVerou** [comment]
I see, thanks! And thanks @latin-1 for the PR. Merging then.

**Pull request merged**